pico/api/src/main/java/io/helidon/pico/DeActivationRequest.java

```diff
@@ -17,41 +17,26 @@
 package io.helidon.pico;
 
 import io.helidon.builder.Builder;
+import io.helidon.common.LazyValue;
 import io.helidon.config.metadata.ConfiguredOption;
 
 /**
- * Request to {@link io.helidon.pico.DeActivator#deactivate(DeActivationRequest)}.
- *
- * @param <T> type to deactivate
+ * Request to deactivate a {@link io.helidon.pico.ServiceProvider}.
  */
 @Builder
-public interface DeActivationRequest<T> {
-
-    /**
-     * Create a request with defaults.
-     *
-     * @param provider service provider responsible for invoking deactivate
-     * @return a new request
-     * @param <T> type to deactivate
-     */
-    @SuppressWarnings("unchecked")
-    static <T> DeActivationRequest<T> create(ServiceProvider<T> provider) {
-        return DefaultDeActivationRequest.builder().serviceProvider(provider).build();
-    }
+public interface DeActivationRequest {
 
     /**
-     * Service provider responsible for invoking deactivate.
-     *
-     * @return service provider
+     * Default request.
      */
-    ServiceProvider<T> serviceProvider();
+    LazyValue<DeActivationRequest> DEFAULT = LazyValue.create(() -> DefaultDeActivationRequest.builder().build());
```

Comment:
This is a publicly accessible constant from anybody. This should not be a LazyValue.
If this is a constant, it should be a real constant. If it needs to be lazy evaluated, create a package local class and expose the value through static method on this interface (LazyValue should be an implementation detail, not public API)

Reply:
good catch. Will make this an abstract class instead, and then hide DEFAULT.

Reply:
changing it from interface to abstract class is not the best approach. It is a very different contract...